### DISCUSSION TITLE
Aggregate functions resulting in zeros when data there are integers in the table.

### DISCUSSION THREAD

**Discussion by kaitlynkinter1** [dbeaver/dbeaver#18992] [Q&A] [answered]
Why would my aggregate function result in zeros when there are numbers(integers) in the data table it is pulling from in dbeaver?
![0DEB7729-88EE-49FC-ADB5-1DC6FD1475C7](https://user-images.githubusercontent.com/116836120/217294771-5e18b56c-475d-4c4b-aa20-d92e77776213.jpeg)

**Comment by E1izabeth** [comment, marked as answer]
It seems that postgresql assume that the result is expected as integer, because both `total_deaths` and `total_cases` are integer types. To handle this you can cast both `total_cases` and `total_deaths` to floating-point type.
Fro example,
this gives 0 `select 1/2;`
but this gives 0.5 `select cast(1 as double precision)/cast(2 as double precision);`

**Comment by kaitlynkinter1** [comment]
Thank you! That is exactly what it was doing!
